content/features.md

```diff
@@ -0,0 +1,30 @@
+# Supported QoS and Features
+
+This table represents the various QoS and features supported by CycloneDDS across the various language implementations.
+
+| QoS / Feature 	| C Support 	| C++ Support 	| Python Support 	| Remarks 	|
+|---	|---	|---	|---	|---	|
+| USER_DATA 	| Yes 	| Yes 	| Yes 	|  	|
+| TOPIC_DATA 	| Yes 	| Yes 	| Yes 	|  	|
+| GROUP_DATA 	| Yes 	| Yes 	| Yes 	|  	|
+| DURABILITY 	| Partial 	| - 	| - 	| only volatile and transient-local 	|
+| DURABILITY_SERVICE 	| Partial 	| - 	| - 	| service_cleanup_delay and resource limits are not supported 	|
+| PRESENTATION 	| Partial 	| - 	| - 	| only the default (INSTANCE, non-coherent, non-ordered) 	|
+| DEADLINE 	| No 	| No 	| No 	|  	|
+| LATENCY_BUDGET 	| Yes 	| Yes 	| Yes 	|  	|
+| OWNERSHIP 	| Yes 	| Yes 	| Yes 	|  	|
+| OWNERSHIP_STRENGTH 	| Yes 	| Yes 	| Yes 	|  	|
+| LIVELINESS 	| Yes 	| Yes 	| Yes 	|  	|
+| TIME_BASED_FILTER 	| Ongoing 	| No 	| No 	| [cyclonedds#1506](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1506) 	|
```

Comment:
This has now been merged. (That was the reason for not reviewing earlier, but I really should have let you know!)

It also should now work on C++ and Python (they simply forward the QoS's to the C library, which implements them).

Reply:
I dont think all features when they're implemented in C flows into C++ and python.
Example: topic_discovery was done with https://github.com/eclipse-cyclonedds/cyclonedds/pull/615 but there is no corresponding PR in the cxx repo
also: from src/ddscxx/src/org/eclipse/cyclonedds/topic/AnyTopicDelegate.cpp
```void
AnyTopicDelegate::discover_topics(
        const dds::domain::DomainParticipant& dp,
        std::vector<dds::topic::TAnyTopic<AnyTopicDelegate> >& topics,
        uint32_t max_size)
{
    (void)dp;
    (void)topics;
    (void)max_size;
    topics.clear();
    ISOCPP_THROW_EXCEPTION(ISOCPP_UNSUPPORTED_ERROR, "Discovery of AnyTopics not implemented yet");
}

Reply:
> I dont think all features when they're implemented in C flows into C++ and python.

That's certainly true. I was simply thinking of supporting a "simple" QoS, where the only interface is the QoS setting itself. "Time based filter" is like that, so then it all falls into place rather easily.

Things like topic discovery are much more involved, they involve "real" implementation work 🙂